libs/expo/betterangels/src/lib/screens/AddEditClient/index.tsx

```diff
@@ -189,8 +229,34 @@ export default function AddEditClient({ id }: { id?: string }) {
 
     const { displayCaseManager, ...updatedClientInput } = data.clientProfile;
 
+    const existingSocialMediaProfiles =
+      data.clientProfile.socialMediaProfiles || [];
+
+    const updatedSocialMediaProfiles = defaultSocialMedias.map(
+      (defaultProfile) => {
+        const existingProfile = existingSocialMediaProfiles.find(
+          (profile) => profile.platform === defaultProfile.platform
+        );
+
+        if (existingProfile) {
+          const { __typename, ...cleanedProfile } = existingProfile;
+          return cleanedProfile;
+        }
+
+        return defaultProfile;
+      }
+    );
+
+    const existingPhoneNumbers =
+      data.clientProfile.phoneNumbers &&
+      data.clientProfile.phoneNumbers.length > 0
+        ? data.clientProfile.phoneNumbers.map(({ __typename, ...rest }) => rest)
+        : [{ number: '', isPrimary: false }];
```

Comment:
if using `existingPhoneNumbers` const, it would probably be more accurate to set `existingPhoneNumbers` to include only data from clientProfile, such as `data.clientProfile.phoneNumbers?.map(.....)`

and then `const phoneNumberEmpyInput: TPhoneInput[] = [{ number: '', isPrimary: false }]` 

finally, below in `clientInput` we can set `phoneNumbers: existingPhoneNumbers || phoneNumberEmpyInput`